apps/prairielearn/src/lib/question.js

```diff
@@ -142,28 +142,36 @@ module.exports = {
    * @param {String} filename
    * @param {Object} variant - The variant.
    * @param {Object} question - The question for the variant.
-   * @param {Object} course - The course for the variant.
+   * @param {Object} variant_course - The course for the variant.
    * @param {string} authn_user_id - The current authenticated user.
    * @param {function} callback - A callback(err, fileData) function.
    */
-  getFile(filename, variant, question, course, authn_user_id, callback) {
+  getFile(filename, variant, question, variant_course, authn_user_id, callback) {
     questionServers.getModule(question.type, (err, questionModule) => {
       if (ERR(err, callback)) return;
-      questionModule.file(filename, variant, question, course, (err, courseIssues, fileData) => {
-        if (ERR(err, callback)) return;
-
-        const studentMessage = 'Error creating file: ' + filename;
-        const courseData = { variant, question, course };
-        module.exports._writeCourseIssues(
-          courseIssues,
+      module.exports._getQuestionCourse(question, variant_course).then((question_course) => {
```

Comment:
This doesn't handle errors properly. If you want to use an async function inside a callback chain like this, I'd recommend `utils.callbackify`.

Reply:
done, thanks